sgkit/io/vcf/vcf_reader.py

```diff
@@ -240,7 +240,7 @@ def for_field(
             dims.append(dimension)
             chunksize += (size,)
 
-        array = np.full(chunksize, fill_value, dtype=dtype)
+        array = np.full(chunksize, missing_value, dtype=dtype)
```

Comment:
I don't understand this change, but I could be missing the overall motivation. The way I think about this is that all positions start as fill (pad) values, then values are filled in with actual or missing values as they are read in (and according to ploidy). 

Perhaps it would help to spell out a few cases that fail. They could well be in the tests, but there are a lot of changes there.

Reply:
Ah yes, this wasn't necessary. fixed.